hw03-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,46 @@
+package homework;
+
+import java.util.*;
+import java.util.Map;
+
+public class CustomerService {
+    private final TreeMap<Customer, String> customers = new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> customers = new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```

Слева тип лучше объявлять интерфейсом, причем минимально необходимым.